src/array.cr

```diff
@@ -836,13 +810,12 @@ class Array(T)
   # a.delete_at(99, 1) # raises IndexError
   # ```
   def delete_at(index : Int, count : Int)
-    index += size if index < 0
-    unless 0 <= index <= size
-      raise IndexError.new
-    end
+    index, count = normalize_start_and_count(index, count)
 
-    val = self[index, count]
-    count = index + count <= size ? count : size - index
+    val = Array(T).build(count) do |buffer|
+      buffer.copy_from(@buffer + index, count)
+      count
+    end
```

Comment:
Yeah this is an unrelated optimization. I am leaving this to a different PR because I realized there are other places in `Array` where the same optimization would apply.